### PR TITLE
fix default_replica_path and default_replica_name values are useless …

### PR DESCRIPTION
**Pull request by mxzlxy**
…on Replicated(*)MergeTree engine when the engine needs specify other parameters

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

fix default_replica_path and default_replica_name values are useless on Replicated(*)MergeTree engine when the engine needs specify other parameters

Detailed description / Documentation draft:

when create ReplicatedMergeTree table, the create clause is: "CREATE TABLE ...... ENGINE =  ReplicatedMergeTree('/clickhouse/tables/{layer}-{shard}/table_name', '{replica}')" , you can omit arguments when specify default arguments(default_replica_path and default_replica_name) in the server configuration file.  but, default arguments are invalid for other replicated engine which has other arguments, for example:ENGINE = ReplicatedReplacingMergeTree(ver) is illegal when we only specify ver.

### CODE REVIEW

File: src/Storages/MergeTree/registerStorageMergeTree.cpp [hunk 1]
```diff
@@ -427,7 +427,7 @@ static StoragePtr create(const StorageFactory::Arguments & args)
                     "No replica name in config" + getMergeTreeVerboseHelp(is_extended_storage_def), ErrorCodes::NO_REPLICA_NAME_GIVEN);
             ++arg_num;
         }
-        else if (is_extended_storage_def && arg_cnt == 0)
+        else if (is_extended_storage_def && (arg_cnt == 0 || !engine_args[arg_num]->as<ASTLiteral>()))
```
Comment (tavplubix) [inline src/Storages/MergeTree/registerStorageMergeTree.cpp:430]:
```suggestion
        else if (is_extended_storage_def && (arg_cnt == 0 || !engine_args[arg_num]->as<ASTLiteral>() || (arg_cnt == 1 && merging_params.mode == MergeTreeData::MergingParams::Graphite)))
```

Extra argument of `ReplicatedGraphiteMergeTree` engine if string literal

**Comment by tavplubix** [comment]
`Fast test — fail: 1` - it was broken in master, let's restart tests

**Comment by mxzlxy** [comment]
@tavplubix Functional stateless tests still failed, how can I fix it, thanks

**Comment by tavplubix** [comment]
Failed tests are known to be flaky, you don't have to do anything

**Review by tavplubix — APPROVED** [review]

**Pull request merged**